openclsim/plugins/weather.py

```diff
@@ -196,3 +208,345 @@ def process_data(self, criterion) -> None:
+
+    An instance of the OffshoreEnvironment class holds information of
+    site specific data. Besides, the class has a certain `find_window`
+    method which searches for the first workable weather window for a
+    given activity.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Class constructor."""
+        super().__init__(*args, **kwargs)
+
+    def store_information(
+        self, var: str = None, value=None, filename: str = None, **kwargs
+    ):
+        """
+        Store metocean information in the dataset.
+
+        The store_information function provides the user a method to
+        pass information to the dataset about site specific metocean
+        conditions. The function takes either a single value using the
+        `value` argument, or a series of values from a `.csv` file. The
+        latter requires two columns of data: (1) timestamps from when
+        the data was recorded and (2) the corresponding values. Besides,
+        the first row of the file is processed as column names.
+
+        If a filename is provided, the pandas package is used to read
+        the contents. For that reason, it is possible to manipulate the
+        reading process using keyword-arguments. Please refer to the
+        `pandas.read_csv` documentation.
+
+        Parameters
+        ----------
+            var: str
+                Variable name as a single word or abbriviation.
+            value: str
+                Value of the corresponding variable.
+            filename: str
+                Path to the `.csv` file.
+
+        Examples
+        --------
+            >>> import openclsim.plugins as plugins
+            >>> oe = plugins.OffshoreEnvironment()
+            >>> # Store a dataset from a *.csv file.
+            >>> oe.store_information(var="hs", filename="./file.csv")
+            >>> # Or store information about, e.g., the sites ID.
+            >>> oe.store_information(var="id", value="Offshore Site ID")
+
+        """
+        # Test if a variable name is (properly) provided.
+        if (var is not None) and (isinstance(var, str) is False):
+            raise ValueError("The `var` argument accepts only a `string` type.")
+        elif var is None:
+            raise TypeError(
+                "`store_information` requires at least the positional arguments: `var`"
+                + " and `value` or `filename`."
+            )
+        else:
+            pass
+
+        # Test if a value or filename is (properly) provided.
+        if (value is None) and (filename is None):
+            raise TypeError(
+                "`store_information` requires at least one of the positional arguments:"
+                + " `value` or `filename` to be set."
+            )
+        elif (value is not None) and (filename is not None):
+            raise TypeError(
+                "`store_information accepts either a `value` or `filename` to be set."
+            )
```

Comment:
Is also a bit new for me but the correct syntaxt is:

```python
assert (value is not None) and (filename is not None), "`store_information accepts either a `value` or `filename` to be set."
```

IMHO this is much cleaner. But as you can see I did not impement this myself in OpenCLSim since I know of it for a week now.